pkg/store/gate.go

```diff
@@ -0,0 +1,54 @@
+package store
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/pkg/gate"
+)
+
+// Gate wraps the Prometheus gate with extra metrics.
+type Gate struct {
+	g              *gate.Gate
+	currentQueries prometheus.Gauge
+	gateTiming     prometheus.Summary
+}
+
+// NewGate returns a new gate.
+func NewGate(maxConcurrent int, reg prometheus.Registerer) *Gate {
+	g := &Gate{
+		g: gate.New(maxConcurrent),
+	}
+	g.currentQueries = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "thanos_bucket_store_queries_total",
```

Comment:
idk if that's only me, but for gauges `thanos_bucket_store_queries_in_flight` sounds a bit better, but up to you guys

Reply:
I agree that we shouldn't use `_total` for gauges.